src/api-v1.js

```diff
@@ -151,9 +151,83 @@ async function validateWorkerType (ctx, workerTypeName, workerType) {
 
 }
 
+/**
+ * Calculate some summary statistics for a worker type, based on the given
+ * WorkerState.
+ */
+function workerTypeSummary(workerType, workerState) {
+  let summary = {
+    workerType: workerType.workerType,
+    minCapacity: workerType.minCapacity,
+    maxCapacity: workerType.maxCapacity,
+    requestedCapacity: 0,
+    pendingCapacity: 0,
+    runningCapacity: 0,
+  };
+
+  let capacities = {};
+  workerType.instanceTypes.forEach(instanceType => {
+    capacities[instanceType.instanceType] = instanceType.capacity;
+  });
+
+  workerState.instances.forEach(instance => {
+    if (instance.state === 'running') {
+      summary.runningCapacity += capacities[instance.type] || 0;
+    } else if (instance.state === 'pending') {
+      summary.pendingCapacity += capacities[instance.type] || 0;
+    } // note that other states are ignored
+  });
+
+  workerState.requests.forEach(request => {
+    summary.requestedCapacity += capacities[request.type] || 0;
+  });
+
+  return summary;
+}
+
 module.exports = api;
 
 api.declare({
+  method: 'get',
+  route: '/worker-types',
+  name: 'listWorkerTypeSummaries', // TODO will become listWorkerTypes after transition
```

Comment:
For authenticated users, there's no reason to calculate the capacities and other status information, and there's no reason to force these calculations if we're only wanting to list the worker types.  This is a lot of extra work if it's not needed.